tests/tracer/test_trace_utils.py

```diff
@@ -446,6 +447,44 @@ def test_set_http_meta_no_headers(mock_store_headers, span, int_config):
     mock_store_headers.assert_not_called()
 
 
+@mock.patch("ddtrace.contrib.trace_utils._store_headers")
+@pytest.mark.parametrize(
+    "user_agent_value, expected_keys ,expected",
+    [
+        ("dd-agent/1.0.0", ["runtime-id", http.USER_AGENT], "dd-agent/1.0.0"),
+        ("ㄲㄴㄷㄸ", ["runtime-id"], None),
+        (u"ㄲㄴㄷㄸ", ["runtime-id", http.USER_AGENT], u"ㄲㄴㄷㄸ"),
+        (
+            None,
+            [
+                "runtime-id",
+            ],
+            None,
+        ),
+        (101234, ["runtime-id"], None),
+        (True, ["runtime-id", http.USER_AGENT], "True"),
+        (False, ["runtime-id"], None),
+        ([], ["runtime-id"], None),
+        ({}, ["runtime-id"], None),
+        (["test1", "test2"], ["runtime-id", http.USER_AGENT], "['test1', 'test2']"),
+        ({"test1": "key1", "test2": "key2"}, ["runtime-id", http.USER_AGENT], "{'test1': 'key1', 'test2': 'key2'}"),
+    ],
+)
+def test_set_http_meta_headers_useragent(
+    mock_store_headers, user_agent_value, expected_keys, expected, span, int_config
+):
+    int_config.myint.http._header_tags = {"enabled": True}
+    assert int_config.myint.is_header_tracing_configured is True
+    trace_utils.set_http_meta(
+        span,
+        int_config.myint,
+        request_headers={"user-agent": user_agent_value},
```

Comment:
do we normalize headers at some point? do all integrations use this casing (or provide case-insensitive access)?